AISKU/src/ApplicationInsightsContainer.ts

```diff
@@ -3,13 +3,14 @@ import { Initialization as ApplicationInsights, Snippet, IApplicationInsights }
 
 export class ApplicationInsightsContainer {
 
-    public static getAppInsights(snippet: Snippet, version: number) : IApplicationInsights | IAppInsightsDeprecated {
-        let initialization = new ApplicationInsights(snippet);
+    public static getAppInsights(snippet: Snippet, version: number = 2.0, applyAppInsights?: () => void) : IApplicationInsights | IAppInsightsDeprecated {
+        const initialization = new ApplicationInsights(snippet);
+        if (applyAppInsights) { applyAppInsights(); }
```

Comment:
what is this for?

Reply:
wondering if this work can be done inside the constructor of Applicationinsights here (new ApplicationInsights(snippet);). Seems it has the object snippet available to setup the fields prior to doing the required initialization through loadAppInsights() which does the emptyQueue which is impacted.

Reply:
It could be done this way too. Is it more confusing if snippet is modified in some other constructor though? This is the only reason I just went with a callback

Reply:
I see, perhaps expose a method called updateSnippetDefs or something like that in Initialization and just call that. If you want to keep as is, thats fine, but update name, I didnt really catch what was going on.

Reply:
updated